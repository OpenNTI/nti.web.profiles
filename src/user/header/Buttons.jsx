import React from 'react';
import PropTypes from 'prop-types';
import {getAppUsername} from '@nti/web-client';

import EditControls from './EditControls';
import ManageControls from './ManageControls';

export default class Buttons extends React.Component {

	static propTypes = {
		entity: PropTypes.object.isRequired
	}

	isMe = () => {
		const {entity} = this.props;

		return entity && getAppUsername() === entity.getID();
	}

	render () {
		const {entity} = this.props;

		return this.isMe() ? <EditControls entity={entity}/> : <ManageControls entity={entity}/>;
	}
}
