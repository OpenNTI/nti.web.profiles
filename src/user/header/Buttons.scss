.profile-edit-controls,
.profile-manage-controls {
	margin-bottom: 1.25rem;
	padding-top: 2.5rem;
	display: flex;
	justify-content: end;

	.nti-button {
		margin-left: 0.625em;
		border-radius: 3px;

		i {
			margin-right: 0.5rem;
		}

		&.follow {
			background-color: var(--primary-green);
			color: white;
		}

		&.unfollow {
			background-color: var(--primary-red);
			color: white;
		}
	}
}

.message-button-flyout {
	color: white;
	font-size: 12px;
	line-height: 19px;
	font-weight: 300;

	.flyout .flyout-arrow::after {
		background-color: #333;
	}

	.flyout-inner {
		font-size: 10px;
		font-weight: 500;
		padding: 4px 8px;
		border-radius: 3px;
		background-color: #333;
		text-transform: uppercase;

		.username {
			margin-right: 3px;
		}
	}
}
