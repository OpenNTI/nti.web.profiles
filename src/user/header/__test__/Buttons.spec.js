import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import Buttons from '../Buttons';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = () => {
	return {
		getContacts: () => {
			return {
				addListener: () => {},
				removeListener: () => {},
				contains: () => true
			};
		}
	};
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(getMockService());
	global.$AppConfig = {
		...(global.$AppConfig),
		username: 'testUser'
	};
};

const onAfter = () => {
	tearDownTestClient();
	delete global.$AppConfig.username;
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe ('User profile header buttons test', () => {
	beforeEach(() => {
		onBefore();
	});

	afterEach(() => {
		onAfter();
	});

	test ('Is not acting user', async () => {
		const user = {
			getID: () => 'NOTtestUser'
		};

		const cmp = renderer.create(<Buttons entity={user}/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
