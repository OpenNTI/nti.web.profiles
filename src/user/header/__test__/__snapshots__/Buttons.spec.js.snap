// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User profile header buttons test Is not acting user 1`] = `
<div
  className="contact profile-manage-controls"
>
  <div
    aria-haspopup="dialog"
    className="flyout-trigger nti-button message message-button-flyout flyout-closed"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span>
      <div
        className="icon"
      />
      <span
        className="text"
      >
        Message
      </span>
    </span>
  </div>
  <a
    className="nti-button unfollow primary"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="content"
    >
      <i
        className="icon-unfollow"
      />
      <span>
        Unfollow
      </span>
    </div>
  </a>
</div>
`;
